test/fakeintake/client/client.go

```diff
@@ -160,8 +159,8 @@ func (c *Client) FilterMetrics(name string, options ...MatchOpt[*aggregator.Metr
 	for _, metric := range metrics {
 		matchCount := 0
 		for _, matchOpt := range options {
-			err = matchOpt(metric)
-			if err != nil {
+			isMatch, err := matchOpt(metric)
+			if !isMatch || err != nil {
```

Comment:
Is there a reason to not return the error?

Reply:
Just that I changed code too quickly ! Changing this, we can indeed return errors right away.